torch_xla/csrc/reduction.cpp

```diff
@@ -545,4 +547,51 @@ std::vector<xla::XlaOp> BuildEinsumBackward(const xla::XlaOp& grad_output,
   return result;
 }
 
+xla::XlaOp BuildMultinomial(xla::XlaOp input, int64_t num_samples, 
```

Comment:
Why is `reduction.cpp` the right place to implement this code. Have you considered `xla_lower_util`?

Reply:
fixed!